### ISSUE TITLE
`oneOfType([Array, any()])` throws error at prop validation

### ISSUE THREAD

**Issue by victorgarciaesgi** [dwightjack/vue-types#140]
Hi it's me again! :D 

I'm translating all my props to vue-types, and came accross a bug with `oneOfType`.

Before:

```ts
props: {
 value: { type: [Array, null] as PropType<any | any[]> }
}
```

So in `vue-types`:

```ts
props: {
 value: oneOfType([Array, any()])
}
```

And I got this errors :(

![image](https://user-images.githubusercontent.com/15092120/125459277-ecf5e6fe-5ee8-4de7-949a-4b3a081ca1d9.png)

**Comment by victorgarciaesgi** [comment]
Ok found out that `{ type: [Array, null] as PropType<any | any[]> }` produce the same error so not `vue-types` fault :p

**Comment by victorgarciaesgi** [comment]
You're right that's exactly the same as putting `any` 😅 so it's my fault here!
Still on Vue 2 (Waiting for Nuxt 3 🙏)
It could be cool to cast to `any()` too, like putting `any<AllPossibleValues>()` or `any<unknown>()`

**Comment by dwightjack** [comment]
Just for reference, there's an open issue about nullable props in vue3 repo as well: https://github.com/vuejs/vue-next/issues/3948.

 Looking at the source for [props validation](https://github.com/vuejs/vue-next/blob/master/packages/runtime-core/src/componentProps.ts#L575) using `null` in an array of types (for example `type: [Array, null]`) might cause some validation errors. 

About casting any, it seems a good idea. I'll take some time tomorrow to think about any side effect it might cause though.

**Comment by dwightjack** [comment]
I published [v4.1.0](https://github.com/dwightjack/vue-types/releases/tag/4.1.0) which should fix this issue.

Closing for now. Feel free to open a new issue if you experience any problems.
